Make MUSCA B1 and MUSCA A1 TF-M ready

Patater opened the pull request:
### Summary of changes <!-- Required -->

Refactor MUSCA platform support in a way that makes the NS binaries compatible with how TF-M expects: same layout in flash and RAM. Doing this doesn't break ordinary Mbed OS usage.

#### Impact of changes <!-- Optional -->
No breaking changes. MUSCA targets will be ready for TF-M.

#### Migration actions required <!-- Optional -->
None

### Documentation <!-- Required -->

None

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

    [X] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

    [] No Tests required for this change (E.g docs only update)
    [X] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

@devran01 @jainvikas8 

----------------------------------------------------------------------------------------------------------------

urutva committed:
musca_a1: Add TF-M compatibility

Make the MUSCA_A1 target TF-M compatible by doing the following:
    - Add flash, region definitions, and preprocessed image macros from
      TF-M (at version 6e7be077eabe "Core: Add lifecycle API")
    - Update the MUSCA_A1 linker script to create a flash image
      compatible with TF-M.
    - Update the tfm/bin_utils/assemble.py signing script to work with
      preprocessed image macros rather than flat C pre-processor defines

Signed-off-by: Devaraj Ranganna <devaraj.ranganna@arm.com>
Signed-off-by: Jaeden Amero <jaeden.amero@arm.com>

Patater committed:
musca: Use the default baud rate

Avoid surprising users. Use the Mbed OS default baud rate. Applications
that want to see TF-M debug prints can both enable and set their baud
rate at the application level. Mbed OS should not turn on TF-M debug
prints nor surprise the user with a non-default baud rate.

Signed-off-by: Jaeden Amero <jaeden.amero@arm.com>

Patater committed:
musca_b1: Add TF-M compatibility

Make the MUSCA_B1 target TF-M compatible by doing the following:
    - Add flash, region definitions, and preprocessed image macros from
      TF-M (at version 6e7be077eabe "Core: Add lifecycle API")
    - Update the MUSCA_B1 linker script to create a flash image
      compatible with TF-M.
    - Update the tfm/bin_utils/assemble.py signing script to work with
      preprocessed image macros rather than flat C pre-processor defines

Signed-off-by: Devaraj Ranganna <devaraj.ranganna@arm.com>
Signed-off-by: Jaeden Amero <jaeden.amero@arm.com>

jainvikas8 commented on targets/TARGET_ARM_SSG/TARGET_MUSCA_A1/TARGET_MUSCA_A1_NS/device/TOOLCHAIN_GCC_ARM/musca_ns.ld:
```diff
@@ -95,25 +78,26 @@ SECTIONS
     .copy.table :
     {
-        . = ALIGN(8);
+        . = ALIGN(4);
```
It would be good to know why the alignment has been changed to 4-bytes in a comment.

jainvikas8 reviewed (changes requested):
Please update the copyright year of the files as needed.

Patater committed:
psa: Include region_defs.h where needed

We are moving to flash_layout.h and region_defs.h being more closely
aligned with TF-M's version. As such, some of the Mbed PSA
implementation needs to additionally include region_defs.h.

adbridge reviewed (approved):
AFAICT looks ok to me
